src/data/prod.lean

```diff
@@ -35,6 +35,15 @@ funext $ map_fst f g
 lemma map_snd' (f : α → γ) (g : β → δ) : (prod.snd ∘ map f g) = g ∘ prod.snd :=
 funext $ map_snd f g
 
+/--
+Composing a `prod.map` with another `prod.map` is equal to
+a single `prod.map` of composed functions.
+-/
+lemma comp_map {ε ζ : Type*}
+  (f : α → β) (f' : γ → δ) (g : β → ε) (g' : δ → ζ) :
+  prod.map g g' ∘ prod.map f f' = prod.map (g ∘ f) (g' ∘ f') :=
```

Comment:
Would this lemma be easier to apply if the function were fully applies? I think there's a very similar list lemma it would be nice to match the name and statement of

Reply:
I think both versions are useful, and it makes sense to include both of them.

Reply:
Do you mean `list.map_map`?

Reply:
Yes!

Reply:
Note we have a lot of `map_map` lemmas: https://leanprover-community.github.io/mathlib_docs/find/map_map

But I can't seem to see any `map_comp_map` lemmas

Reply:
I'm working on maps on top of definitions which wrap `list.zip`, and it's easier to show, without a congr step, that the two maps are equal than to show that the result of their application.